### ISSUE TITLE
Reexport jl_arrayref, since it's still there, and broke at least one package

### ISSUE THREAD

**Issue by PallHaraldsson** [JuliaLang/julia#56805]
Memory PR unexported it and proke Moustrap.jl, which is unregistered, and possibly 2/3 other:

https://juliahub.com/ui/Search?q=jl_array&type=packages

The search may be unreliable, it told me also used in ArrayAllez.jl but it at least passed tests for me. I also don't see where it's used in Mousetrap.jl or if more needs to be added/reexported:

https://www.reddit.com/r/Julia/comments/1h27zay/the_mousetrap_is_a_trap/

**Comment by adienes** [comment]
![Image](https://github.com/user-attachments/assets/da8eec53-d46f-4a76-8d5b-5b27e0968f8e)
probably not a good idea.

**Comment by PallHaraldsson** [comment]
Where are you searching, I find 9 results now including in Knet.jl: https://juliahub.com/ui/Search?q=jl_arrayref&type=code Not sure what I was finding before...

**Comment by oscardssmith** [comment]
why are they ccalling rather than just using `getindex`?

**Comment by PallHaraldsson** [comment]
Not sure, I'm just looking into what broke and where. Packages could be "fixed", but is it better to reexport? Do you know if it's enough, and the definition is equivalent? There might be way more needing fixing in the package[s], or not, I'm just starting investigating...

**Comment by adienes** [comment]
I'm searching in the docs. I don't think this has ever been anything remotely close to public api...

**Comment by PallHaraldsson** [comment]
Fair enough, still broke a package, and if not reexporting, then would like to know best way forward.

What exactly is the "Julia C API"? https://docs.julialang.org/en/v1/manual/embedding/

Is it nothing, expect what's documented (on that page)? It doesn't say stable or unstable or exported. So is there some list of approved (or unapproved, or risky)?

I think the files controlling exporting with previously:
>XX(jl_arraylen) \

only mean that you  CAN/COULD call, not that you should.

I tried to look up [un]stable in the docs and found "unstable" here, and:
https://docs.julialang.org/en/v1/devdocs/builtins/
>Core..memoryrefoffset

That's probably a typo .. -> .

FYI: We have a project like unoffical:
https://github.com/Suzhou-Tongyuan/TyJuliaCAPI.jl
>Stable and generic C API for Julia

**Comment by PallHaraldsson** [comment]
I see it can't probably be reexported since nowe:
``` 
- JL_DLLEXPORT size_t jl_array_size(jl_value_t *a, int d)
+ JL_DLLEXPORT size_t jl_array_size(jl_array_t *a, int d)
  {
+     // n.b this functions only use was to violate the vector abstraction, so we have to continue to emulate that
+     if (d >= jl_array_ndims(a))

+        return a->ref.mem->length;
      return jl_array_dim(a, d);
  }
```
